backend/src/apiserver/resource/resource_manager.go

```diff
@@ -130,63 +130,71 @@ func (r *ResourceManager) getScheduledWorkflowClient(namespace string) scheduled
 	return r.swfClient.ScheduledWorkflow(namespace)
 }
 
-func (r *ResourceManager) GetTime() util.TimeInterface {
-	return r.time
-}
-
-func (r *ResourceManager) CreateExperiment(inputExperiment interface{}) (*model.Experiment, error) {
-	experiment, err := r.ToModelExperiment(inputExperiment)
-	if err != nil {
-		return nil, util.Wrap(err, "Failed to convert experiment model")
-	}
+// Creates a new experiment.
+func (r *ResourceManager) CreateExperiment(experiment *model.Experiment) (*model.Experiment, error) {
 	return r.experimentStore.CreateExperiment(experiment)
 }
 
+// Fetches an experiment with the given id.
 func (r *ResourceManager) GetExperiment(experimentId string) (*model.Experiment, error) {
 	return r.experimentStore.GetExperiment(experimentId)
 }
 
-func (r *ResourceManager) ListExperiments(filterContext *common.FilterContext, opts *list.Options) (
-	experiments []*model.Experiment, total_size int, nextPageToken string, err error) {
+// Fetches experiments with the given filtering and listing options.
+func (r *ResourceManager) ListExperiments(filterContext *model.FilterContext, opts *list.Options) ([]*model.Experiment, int, string, error) {
 	return r.experimentStore.ListExperiments(filterContext, opts)
 }
 
-func (r *ResourceManager) DeleteExperiment(experimentID string) error {
-	_, err := r.experimentStore.GetExperiment(experimentID)
+// Deletes the experiment with the given id.
+func (r *ResourceManager) DeleteExperiment(experimentId string) error {
+	_, err := r.experimentStore.GetExperiment(experimentId)
 	if err != nil {
-		return util.Wrap(err, "Delete experiment failed")
+		return util.Wrapf(err, "Failed to delete experiment %v due to error fetching it", experimentId)
+	}
+	defaultExperimentId, err := r.GetDefaultExperimentId()
+	if err != nil {
+		return util.Wrapf(err, "Failed to delete experiment %v due to error fetching the default experiment id", experimentId)
+	}
+	if defaultExperimentId != "" && experimentId == defaultExperimentId {
+		return util.NewBadRequestError(util.NewInvalidInputError("Experiment id cannot be equal to the default id %v", defaultExperimentId), "Failed to delete experiment %v. The default experiment cannot be deleted", experimentId)
```

Comment:
(more of a note to myself)

Question: in the old v1 behavior, if users has a system-created default experiment and a user-created custom experiment, when they delete the system-created default experiment, would the custom experiment be promoted as "Default"?
No action needed at this moment, but maybe after beta-0, we may revisit if we can keep the v1 behavior here.

Reply:
In v1.8, when the default experiment gets deleted, we [call UnsetDefaultExperimentIdIfIdMatches](https://github.dev/kubeflow/pipelines/blob/0c77b78899240603cc63b053f66fbb7525cce4d6/backend/src/apiserver/storage/experiment_store.go#L233). No experiment gets promoted to the default. 

IMO, the default experiment should be created at a startup and should be kept the same. This will make it easier to maintain the old behavior after migration. When we revisit this, we may need to clean up this logic as it contains unnecessary function calls.